### PR TITLE
nls web story

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
The idea here is that we use the unpkg service to unpack and provide the AMD modules that Core can require.

Pairs with: https://github.com/microsoft/vscode-unpkg/pull/40 and https://github.com/microsoft/vscode-loader/pull/38

### CODE REVIEW

**Comment by joaomoreno** [comment]
@TylerLeonhardt Seems like the tests trip:

image

File: src/vs/code/browser/workbench/workbench-dev.html [hunk 1]
```diff
@@ -52,7 +57,12 @@
 					throw new Error(`Invalid script url: ${value}`)
 				}
 			}),
-			paths: self.webPackagePaths
+			paths: self.webPackagePaths,
+			'vs/nls': {
+				availableLanguages: {
+					'*': navigator.language
```
Comment (alexdima) [inline src/vs/code/browser/workbench/workbench-dev.html:63]: We should probably limit this to the set of languages we have.

File: src/vs/code/browser/workbench/workbench-dev.html [hunk 2]
```diff
@@ -41,6 +41,11 @@
 		Object.keys(self.webPackagePaths).map(function (key, index) {
 			self.webPackagePaths[key] = `${baseUrl}/remote/web/node_modules/${key}/${self.webPackagePaths[key]}`;
 		});
+
+		if (!navigator.language.startsWith('en')) {
+			self.webPackagePaths[`vs/workbench/workbench.web.main.nls.${navigator.language}`] = `{{WORKBENCH_NLS_BASE_URL}}/vs/code/browser/workbench/workbench.nls.${navigator.language}.js`;
```
Comment (alexdima) [inline src/vs/code/browser/workbench/workbench-dev.html:46]: We might have other entry points that we'd need to cover here. Given our Slack discussion, maybe an easier path would be to have an anonymous `define` call inside `workbench.nls.LANG.js` and then the nls plugin can load it with any name it wants.

**Review by alexdima — DISMISSED** [review]

**Comment by TylerLeonhardt** [comment]
I'm leaving this in draft until the unpkg PR goes in.. but @joaomoreno @alexdima you're welcome to review.

File: src/vs/server/node/webClientServer.ts [hunk 3]
```diff
@@ -340,7 +342,7 @@ export class WebClientServer {
 			'default-src \'self\';',
 			'img-src \'self\' https: data: blob:;',
 			'media-src \'self\';',
-			`script-src 'self' 'unsafe-eval' ${this._getScriptCspHashes(data).join(' ')} 'sha256-fh3TwPMflhsEIpR8g1OYTIMVWhXTLcjQ9kh2tIpmv54=' http://${remoteAuthority};`, // the sha is the same as in src/vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html
+			`script-src 'self' 'unsafe-eval' ${this._getScriptCspHashes(data).join(' ')} 'sha256-fh3TwPMflhsEIpR8g1OYTIMVWhXTLcjQ9kh2tIpmv54=' http://${remoteAuthority} ${nlsBaseUrl ?? ''};`, // the sha is the same as in src/vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html
```
Comment (TylerLeonhardt) [inline src/vs/server/node/webClientServer.ts:345]: Is this the right practice?

File: src/vs/nls.js [hunk 4]
```diff
@@ -148,7 +149,8 @@ var NLSLoaderPlugin;
                     });
                 }
                 else {
-                    req([name + suffix], messagesLoaded_1);
+                    var base = (_a = pluginConfig.baseUrl) !== null && _a !== void 0 ? _a : '';
```
Comment (TylerLeonhardt) [inline src/vs/nls.js:152]: @alexdima do I revert this change and wait until the vscode-loader PR goes in and gets added to vscode?

Comment (TylerLeonhardt) [reply src/vs/nls.js]: updated loader in core already
